source/common/http/conn_manager_impl.cc

```diff
@@ -840,7 +840,6 @@ void ConnectionManagerImpl::ActiveStream::decodeHeaders(RequestHeaderMapPtr&& he
   filter_manager_.requestHeadersInitialized();
   if (request_header_timer_ != nullptr) {
     request_header_timer_->disableTimer();
-    request_header_timer_.reset();
```

Comment:
Why was this reset removed?

Reply:
My test was hitting a use-after-free. I'm going to address this in my cleanup for all of these disable-then-resets.

Reply:
Keeping this timer around for a long time wasting memory is not great.  Can you address the issue in this PR?

Reply:
Done.